### PR TITLE
Team 1 menus

### LINKED ISSUE #4: Main/Pause Menus
# Description
Create intuitive game menus for the start screen and game pausing. The menus should allow the user to easily start or resume a game, configure settings and navigate through different points in the game, such as the home screen, different levels and different gamemodes. The menu buttons should be named and designed so their function is clear to first-time users.
In the final product, the start screen menu will appear at the start of the game and the pause menu will appear when the user pauses the game.
Both screens should aim to be visually and emotionally appealing, especially the start screen as it is the user's first impression of the game.

### Main menu
Displayed upon game startup. This screen will have a background image that fits the game’s theme and displays the game’s logo.
- **Start new game**: Go to opening cutscene with minimal wait time.
  - Difficulty selection: Buttons to choose a difficulty for the game.
- **Resume game**: Shown only when the user has a game in progress. On click, resume game with minimal wait time.
- **How to play**: Guide to how to play the game. List the keys that are used and what each one does (no mouse). Give an overview of how to play the game. This will most likely be a static screen at first, but it can be interactive later.
- **Game settings**: Volume/sound effects sliders, customisable keybindings.
- **Achievements**: Optional. Could show stats and levels reached.

### Pause menu
Displayed when the user pauses by pressing `P`.
- **Save and quit**: Return to the main menu.
- **Resume**: Go back to the game.
- **How to play**: Same as with the main menu. This could be useful if the player forgets how to play or jumps right into a game without checking how to play first.

# Dependencies
- #11 

# Tasks
- [ ] #22 
- [ ] #20 
- [ ] #21 
- [ ] #25 

# Milestones
- [ ] Design the initial layout and navigation structure of the game menu (Aug. 15)
- [ ] Implement pause, continue, and save functionalities within the menu (Aug. 20)
- [ ] Implement the game's settings page (Aug.23)
- [ ] Develop dynamic menu options based on game state and player interactions (Aug. 26)
- [ ] Finalise documentation (Aug. 29)

**Completion Deadline:** 29 Aug

# Extras (optional)
- [ ] How to Play? : Add a section with tutorials or guides that explain the rules, controls, and tips for the game, helping new players get started.
- [ ] Character List Menu: In case we have many character options, there should be a menu to display all locked and unlocked characters.

# Documentation
- [Menu design](https://github.com/UQcsse3200/2024-studio-1/wiki/Menu-design)
- [Menu features and code documentation](https://github.com/UQcsse3200/2024-studio-1/wiki/Menu)
- [User guide to starting the game](https://github.com/UQcsse3200/2024-studio-1/wiki/Starting-the-game)

# Members
- Jigyasu Mittal (@JigyasuMittal )
- Sahajdeep Singh Narang (@sahajdeep31)
- Ian Pinto (@IanP96)
- Tri Nhan Pham (@TriNhan-Pham)
- Aditya Raj Singh (@Aditya6283)

### PR DESCRIPTION
**Pull request by IanP96**
Closes #4 

# Features
- Redesign main menu (bg, fonts, colours, main screen icon).
- New buttons on the main menu
  - how to play (shows static how-to-play guide taken from wiki)
  - Difficulty selection (easy, medium, hard). This difficulty is added as a field in `GdxGame`.
- Settings screen has new options such as mute, sound volume, music volume

The *pause menu* will be implemented for the next sprint.

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java [hunk 1]
```diff
@@ -90,6 +91,8 @@ public void create() {
         spawnMinotaur();
         spawnGhostKing();
 
+        playMusic();
```
Comment (vsinh14) [inline source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:94]: Remove music

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java [hunk 2]
```diff
@@ -239,35 +242,40 @@ private void spawnShieldPotion() {
...
-        music.setVolume(0.3f);
-        music.play();
+  private void playMusic() {
+    Music music = ServiceLocator.getResourceService().getAsset(backgroundMusic, Music.class);
+    music.setLooping(true);
+    if(!UserSettings.get().mute)
+    {
+      music.setVolume(UserSettings.get().musicVolume);
     }
+    else {
```
Comment (vsinh14) [inline source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:274]: no need dont initialise music

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java (hunk 1 above)
Comment (IanP96) [reply source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:94]: We added muting and music volume controls, that's why we added it back in

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java (hunk 2 above)
Comment (IanP96) [reply source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:274]: Same as https://github.com/UQcsse3200/2024-studio-1/pull/97/files/c3b6303b21f379f17266da5f1779a1c08e1708da#r1733918999

**Pull request merged**